From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MrPowerGamerBR <git@mrpowergamerbr.com>
Date: Fri, 17 Nov 2023 14:22:41 -0300
Subject: [PATCH] SparklyPaper: Skip "MapItem#update()" if the map does not
 have the CraftMapRenderer present

Optimizes "image in map" maps, without requiring the map to be locked, which some old map plugins may not do

This has the disadvantage that the vanilla map data will never be updated while the CraftMapRenderer is not present, but that's not a huuuge problem for us

diff --git a/src/main/java/net/minecraft/world/item/MapItem.java b/src/main/java/net/minecraft/world/item/MapItem.java
index 58f3c9598301a5b34142ddf958aa7a4db634be3d..c4c1e922c14bd6db74d08ecf198f4b165845fcdc 100644
--- a/src/main/java/net/minecraft/world/item/MapItem.java
+++ b/src/main/java/net/minecraft/world/item/MapItem.java
@@ -280,9 +280,9 @@ public class MapItem extends Item {
                     mapItemSavedData.tickCarriedBy(player, stack);
                 }
 
-                if (!mapItemSavedData.locked && (selected || entity instanceof Player && ((Player)entity).getOffhandItem() == stack)) {
-                    this.update(world, entity, mapItemSavedData);
-                }
+                    if (!mapItemSavedData.locked && (!one.tranic.sewlia.config.mod.optimization.map.SkipUpdateIfNoRenderer.value || mapItemSavedData.mapView.getRenderers().stream().anyMatch(mapRenderer -> mapRenderer.getClass() == org.bukkit.craftbukkit.map.CraftMapRenderer.class)) && (selected || entity instanceof Player && ((Player)entity).getOffhandItem() == stack)) { // SparklyPaper - don't update maps if they don't have the CraftMapRenderer in the render list
+                        this.update(world, entity, mapItemSavedData);
+                    }
                 } // Folia - region threading
             }
         }
